src/core/meshcop/dataset_manager.cpp

```diff
@@ -505,7 +505,7 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
 
             case Tlv::kDelayTimer:
             {
-                DelayTimerTlv *delayTimerTlv = reinterpret_cast<DelayTimerTlv *>(&tlv);
+                DelayTimerTlv *delayTimerTlv = reinterpret_cast<DelayTimerTlv *>(&data);
```

Comment:
seems need to update the wait time in certificate script as well to wait the pending dataset takes effect. For 9.2.11 and 9.2.18, since affecting new master key, the delay timer is set to default value(300s); For 9.2.7, the delay timer is set to minimal value(30s).